tests/models/test_transformers.py

```diff
@@ -58,27 +58,85 @@ def test_transformer_smoke_select(model_name, model_kwargs):
 
 
 @pytest.mark.skip("Don't overload the build machines")
-def test_phi3_loading():
+def test_phi3_transformers_orig():
+    import torch
+    from transformers import AutoModelForCausalLM, pipeline, AutoTokenizer
+
+    torch.random.manual_seed(0)
+    model = AutoModelForCausalLM.from_pretrained(
+        "microsoft/Phi-3-mini-4k-instruct", 
+        device_map="mps",
+        trust_remote_code=True,
+    )
+
+    tokenizer = AutoTokenizer.from_pretrained("microsoft/Phi-3-mini-4k-instruct")
+    pipe = pipeline(
+        "text-generation",
+        model=model,
+        tokenizer=tokenizer,
+    )
+
+    generation_args = {
+        "max_new_tokens": 5,
+        "return_full_text": True,
+        "temperature": 0.0,
+        "do_sample": False,
+    }
+
+    input_text = "You are a counting bot. Just keep counting numbers. 1,2,3,4"
+    output = pipe(input_text, **generation_args)
+    assert "5" in (output[0]['generated_text'])
+
 
+@pytest.mark.skip("Don't overload the build machines")
+def test_phi3_loading():
     lm = models.Transformers(
         r"microsoft/Phi-3-mini-4k-instruct", trust_remote_code=True
     )
-    lm += f"""Finish counting to 5: 1,2,3,4, + {gen("five", max_tokens=1)}"""
-    assert lm["five"] == "5"
+    lm += f"""You are a counting bot. Just keep counting numbers. 1,2,3,4, <|assistant|>"""
+    lm += gen("five", max_tokens=10)
+    assert "5" in lm["five"]
 
 
 @pytest.mark.skip("Don't overload the build machines")
-def test_phi3_chat():
-    # TODO: Double check chat format: https://huggingface.co/microsoft/Phi-3-mini-4k-instruct
-
-    lm = models.TransformersChat(
-        r"microsoft/Phi-3-mini-4k-instruct", trust_remote_code=True
+def test_llama3_chat():
+    lm = models.Transformers(
+        r"meta-llama/Meta-Llama-3-8B-Instruct", trust_remote_code=True
     )
     with system():
         lm += "You are a counting bot. Just keep counting numbers."
     with user():
         lm += "1,2,3,4"
     with assistant():
-        lm += gen(name="five", max_tokens=1)
+        lm += gen(name="five", max_tokens=10)
+
+    assert "5" in lm["five"]
+
+
+@pytest.mark.skip("Don't overload the build machines")
```

Comment:
I'm getting nervous about the number of tests added which are also skipped. Can you do something like:

https://github.com/guidance-ai/guidance/blob/13270bf9d1d67c3978384cf4df394fb2dfca76f4/tests/models/test_llama_cpp.py#L10-L15

So that these tests will run whenever there's a Phi3 model active in `selected_model` ?